### PR TITLE
Activity card

### PR DESCRIPTION
**Pull request by qwangd2l**

### CODE REVIEW

**Comment by qwangd2l** [comment]
> We will want to make this use the entity store at some point. However, we are not using it yet for the activity list item. So I think when we switch that one we will switch this one.

Does this involve using `BrightspaceHypermediaComponents/organizations#ssemver:^3` where a token is needed but not a presentation href?

**Comment by m6jones** [comment]
Yes, and you would extend the behaviour to get the fetches for free.

File: components/d2l-activity-card/d2l-activity-card.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	}
+	connectedCallback() {
+		super.connectedCallback();
+		const image = this.shadowRoot.querySelector('d2l-course-image');
+		if (image) {
+			image.addEventListener('course-image-loaded', this._activityImageLoaded.bind(this));
+		}
+		this.addEventListener('d2l-organization-accessible', this._onD2lOrganizationAccessible);
+	}
+	disconnectedCalledback() {
```
Comment (AdmiralCoco) [inline components/d2l-activity-card/d2l-activity-card.js]: Typo: `disconnectedCallback`
I wonder if was this working as expected? 😛

File: components/d2l-activity-card/d2l-activity-card.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+				this._image = imageEntity;
+			}
+		}
+
+		return Promise.resolve();
+	}
+	_onD2lOrganizationAccessible(e) {
+		if (e && e.detail && e.detail.organization) {
+			if (e.detail.organization.name) {
+				this._accessibilityData.organizationName = e.detail.organization && e.detail.organization.name;
```
Comment (AdmiralCoco) [inline components/d2l-activity-card/d2l-activity-card.js]: Extra check here for `e.detail.organization` isn't needed

File: components/d2l-activity-card/d2l-activity-card.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+			}
+			if (e.detail.organization.code) {
+				this._accessibilityData.organizationCode = e.detail.organization.code;
+			}
+		}
+		if (e.detail.semesterName) {
+			this._accessibilityData.semesterName = e.detail.semesterName && e.detail.semesterName;
```
Comment (AdmiralCoco) [inline components/d2l-activity-card/d2l-activity-card.js]: Unnecessary check here also

File: components/d2l-activity-card/d2l-activity-card.js (hunk 1 above)
Comment (qwangd2l) [reply components/d2l-activity-card/d2l-activity-card.js]: Did not test disconnected :(
Good catch!

**Pull request merged**